packages/fuels-accounts/src/provider.rs

```diff
@@ -299,41 +303,113 @@ impl Provider {
         Ok(self.client.node_info().await?.into())
     }
 
-    pub async fn checked_dry_run<T: Transaction>(&self, tx: T) -> Result<TxStatus> {
-        let receipts = self.dry_run(tx).await?;
-        Ok(Self::tx_status_from_receipts(receipts))
+    pub async fn latest_gas_price(&self) -> Result<LatestGasPrice> {
+        Ok(self.client.latest_gas_price().await?)
     }
 
-    fn tx_status_from_receipts(receipts: Vec<Receipt>) -> TxStatus {
-        let revert_reason = receipts.iter().find_map(|receipt| match receipt {
-            Receipt::ScriptResult { result, .. } if *result != ScriptExecutionResult::Success => {
-                Some(format!("{result:?}"))
-            }
-            _ => None,
-        });
-
-        match revert_reason {
-            Some(reason) => TxStatus::Revert {
-                receipts,
-                reason,
-                revert_id: 0,
+    pub async fn estimate_gas_price(&self, block_horizon: u32) -> Result<EstimateGasPrice> {
+        Ok(self.client.estimate_gas_price(block_horizon).await?)
+    }
+
+    pub async fn checked_dry_run(&self, tx: impl Transaction) -> Result<TxStatus> {
+        let [(_, tx_status)] = self
+            .client
+            .dry_run(Transactions::new().insert(tx).as_slice())
+            .await?
+            .into_iter()
+            .map(Self::tx_status_from_execution_status)
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("should have only one element");
+
+        Ok(tx_status)
+    }
+
+    pub async fn checked_dry_run_multiple(
+        &self,
+        transactions: Transactions,
+    ) -> Result<Vec<(TxId, TxStatus)>> {
+        Ok(self
+            .client
+            .dry_run(transactions.as_slice())
+            .await?
+            .into_iter()
+            .map(Self::tx_status_from_execution_status)
+            .collect())
+    }
+
+    fn tx_status_from_execution_status(
+        tx_execution_status: TransactionExecutionStatus,
+    ) -> (TxId, TxStatus) {
+        (
+            tx_execution_status.id,
+            match tx_execution_status.result {
+                TransactionExecutionResult::Success { receipts, .. } => {
+                    TxStatus::Success { receipts }
+                }
+                TransactionExecutionResult::Failed { receipts, result } => TxStatus::Revert {
+                    reason: TransactionExecutionResult::reason(&receipts, &result),
+                    receipts,
+                    revert_id: 0,
+                },
             },
-            None => TxStatus::Success { receipts },
-        }
+        )
     }
 
-    pub async fn dry_run<T: Transaction>(&self, tx: T) -> Result<Vec<Receipt>> {
-        let receipts = self.client.dry_run(&tx.into()).await?;
+    pub async fn dry_run(&self, tx: impl Transaction) -> Result<Vec<Receipt>> {
```

Comment:
We can remove this one since it doesn't provide anything compared to `checked_dry_run` where the receipts are contained in the status already.
We don't need the `checked` prefix then either.

Reply:
Makes sense